library/std/src/sys_common/alloc.rs

```diff
@@ -33,14 +33,25 @@ pub unsafe fn realloc_fallback(
     old_layout: Layout,
     new_size: usize,
 ) -> *mut u8 {
-    // Docs for GlobalAlloc::realloc require this to be valid:
-    let new_layout = Layout::from_size_align_unchecked(new_size, old_layout.align());
+    // SAFETY: as stated in docs for GlobalAlloc::realloc, the caller
+    // must guarantee that `new_size` is valid for a `Layout`.
```

Comment:
Please put any guarantees the caller must make in a documentation comment on the function.